A number of small lints

nyurik opened the pull request:
* disable `clippy::too_many_arguments` -- triggers too many times, not worth it
* some useless conversions and other things clippy spotted
* a few spelling things

nyurik committed:
A number of small lints

* disable `clippy::too_many_arguments` -- triggers too many times, not worth it
* some useless conversions and other things clippy spotted
* a few spelling things

nyurik committed:
A number of small lints

* disable `clippy::too_many_arguments` -- triggers too many times, not worth it
* some useless conversions and other things clippy spotted
* a few spelling things

nyurik commented:
CI fails due to a breaking change in zip2 - see https://github.com/zip-rs/zip2/issues/333

usamoi commented:
`zip-extract v0.2.2` has fixed the issue, just one hour after you discovered it.

nyurik committed:
fix pg12 and tests

eeeebbbbrrrr committed:
Merge branch 'develop' into lints

eeeebbbbrrrr committed:
remove some leftover pg12 files (#2036)

eeeebbbbrrrr commented:
@nyurik thanks for this.  I had to rebase it to remove a few pg12 references, and that has ended up in PR #2037.  I'll make sure you're referenced in the commit.  Sorry 'bout that.

usamoi committed:
include `utils/spccache.h` (#2023)

https://github.com/postgres/postgres/blob/master/src/include/utils/spccache.h

It's used to get `spc_random_page_cost`, `spc_seq_page_cost`,
`io_concurrency` and `maintenance_io_concurrency` for a table.

eeeebbbbrrrr committed:
remove pg12 references

eeeebbbbrrrr commented:
got *this* PR rebased and updated to remove the pg12 references.  merging now

nyurik commented:
Thx!
